compositor_render/src/scene/rescaler_component.rs

```diff
@@ -106,27 +106,33 @@ impl RescalerComponent {
         // TODO: to handle cases like transition from top to bottom this view needs
         // to be further processed to use the same type of coordinates as end
         let start = previous_state.map(|state| state.transition_snapshot(ctx.last_render_pts));
+        let end = RescalerComponentParam {
+            id: self.id,
+            position: self.position,
+            mode: self.mode,
+            horizontal_align: self.horizontal_align,
+            vertical_align: self.vertical_align,
+            border_radius: self.border_radius,
+            border_width: self.border_width,
+            border_color: self.border_color,
+            box_shadow: self.box_shadow,
+        };
+
+        let props_changed = previous_state.map(|state| state.end != end).unwrap_or(true);
+        let should_reset_transition = self.transition.map(|t| t.reset_on_update).unwrap_or(false);
```

Comment:
misleading name, it implies whether transition should be reset while it is still depends on props_changed value